### PR TITLE
feat: tsup

### LINKED ISSUE #22: Moving from rollup -> tsup increases bundle size
**Version:** 1.2.0-beta.1

**Describe the bug**
Reverting the change for now as moving from rollup -> tsup increases bundle size a lot. Check the screenshot for reference

Screenshot 2023-03-24 at 1 37 13 PM

Screenshot 2023-03-24 at 1 37 06 PM

**Steps To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by productdevbook**
close #22

### CODE REVIEW

**Comment by lakhansamani** [comment]
@productdevbook if you see the file size of index.global.js its 816KB which is quite some high

Screenshot 2023-03-25 at 10 29 17 AM

File: package.json [hunk 1]
```diff
@@ -1,23 +1,36 @@
 {
 	"version": "1.2.1",
 	"license": "MIT",
-	"types": "lib/cjs/index.d.ts",
-	"main": "lib/cjs/index.js",
-	"module": "lib/esm/index.js",
+	"typings": "dist/index.d.ts",
```
Comment (lakhansamani) [inline package.json:4]: this should be `types` and `typings` else package where this library is used will throw error for typescript definition missing

**Comment by productdevbook** [comment]
If you examine the two files, it will add the codes that should be here. Your old build doesn't have them.


old code empty ? There seems to be an error on how to import it while using it in the browser. The current structure is correct.

image

File: package.json (hunk 1 above)
Comment (productdevbook) [reply package.json]: fixed

**Comment by lakhansamani** [comment]
@productdevbook unpkg / cdn versions are useful for browser only and browser does not need `cross-fetch`
that is the reason I have this code which will create optimized build for browser:
[crossFetch](https://github.com/authorizerdev/authorizer-js/blob/414e7686b2587c03e10b2d9a17b8f38ec9b4508f/src/index.ts#L20-L21)

**Comment by productdevbook** [comment]
check please fixed and name change `authorizer.min.js`

**Comment by lakhansamani** [comment]
Thanks @productdevbook this seems much more optimized 👍

**Pull request merged**